src/audio/SDL_audiocvt.c

```diff
@@ -233,6 +237,16 @@ void ConvertAudio(int num_frames, const void *src, SDL_AudioFormat src_format, i
         src = buf;
     }
 
+    // Gain adjustment
+    if (gain != 1.0f) {
+        float *buf = (float *) (dstconvert ? scratch : dst);
+        const int total_samples = num_frames * src_channels;
+        for (int i = 0; i < total_samples; i++) {
+            buf[i] *= gain;
```

Comment:
I think this should be `buf[i] = src[i] * gain`? Otherwise you aren't actually reading from the current src buffer.

Reply:
Ah, you're right...I'd been hitting codepaths where src was already assigned to a destination buffer from a previous conversion. Fixed in 20bc1573ea3d505924d7e967ac4f9b43980c7e6c.